trees.c

```diff
@@ -670,7 +670,7 @@ void Z_INTERNAL zng_tr_flush_block(deflate_state *s, char *buf, uint32_t stored_
                 opt_lenb, s->opt_len, static_lenb, s->static_len, stored_len,
                 s->sym_next / 3));
 
-        if (static_lenb <= opt_lenb)
+        if (static_lenb <= opt_lenb || s->strategy == Z_FIXED)
```

Comment:
The commits from madler/zlib where these changes are made should be copied over, instead of a single lump-sum commit.

Reply:
I was going to do that after GitHub CI starts working again... It had quite long backlog... I just wanted to do one CI run that actually verifies that there was no ABI breakage between 1.2.12 and 1.2.13...

Reply:
Now there is still one commit that glues things together, but I cherry-picked and cleaned the original commits for bug fixes from madler/zlib...